### PR TITLE
QUARKUS-3290 - Generate source zip which contains the source code of …

### PR DESCRIPTION
**Pull request by rsvoboda**
QUARKUS-3290 - Generate source zip which contains the source code of the built artifacts

JIRA: https://issues.redhat.com/browse/QUARKUS-3290

Quarkus documentation: not applicable

 - [x] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-3290.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * Evaluation of existing test scenarios
+ * New test development
+
+Topics for exploratory testing:
+ * SRC ZIP naming
+ * Root directories naming
+ * Sources for platform participants
+ * Review of included sources for transitive dependencies
+ * Files count, summary per file extension
+ * Large files check
```
Comment (fedinskiy) [inline QUARKUS-3290.md:22]: Just curious: why check large files?

File: QUARKUS-3290.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# QUARKUS-3290 - Generate source zip which contains the source code of the built artifacts
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-3290
+
+As part of discussions abound QUARKUS-2701, it was decided to generate source zip which contains the source code of the artifact from the internal Gerrit repository for upcoming 3.x release
```
Comment (fedinskiy) [inline QUARKUS-3290.md:5]: "The artifact" means our main zip for binaries?

File: QUARKUS-3290.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+One of the reasons is to fully comply with https://issues.redhat.com/secure/attachment/12796750/screenshot-1.png from Red Hat compliance training
```
Comment (fedinskiy) [inline QUARKUS-3290.md:7]: Do we have this text in writing somewhere? Link to jira attachment with text isn't the best solution

**Review by fedinskiy — COMMENTED** [review]
Just a couple of questions

File: QUARKUS-3290.md (hunk 1 above)
Comment (rsvoboda) [reply QUARKUS-3290.md:22]: e.g. https://issues.redhat.com/browse/QUARKUS-3293 ;)

File: QUARKUS-3290.md (hunk 2 above)
Comment (rsvoboda) [reply QUARKUS-3290.md:5]: Should be `artifact => Maven artifacts`

File: QUARKUS-3290.md (hunk 3 above)
Comment (rsvoboda) [reply QUARKUS-3290.md:7]: No text available that could be linked.

I'm not sure I should paste full trext into the test plan, that's why I linked the JIRA attachment.

```
Complete Corresponding Source Code
Complete Corresponding Source Code must include whatever is reasonably necessary for a package to be rebuilt
from source by a technically-skilled recipient with access to an appropriate build environment, without engaging in
excessive effort or experimentation. Items such as Makefiles, build and install scripts, build instructions, RPM spec
files, and POM files must be included with the source code as appropriate. Note that there are separate
requirements to make source code publicly available for Red Hat products for purposes of meeting our obligations
under the US and other export laws that may exceed or differ from the requirements described here. Please review
this link for more information on these separate requirements relating to export laws: separate requirements.
```

File: QUARKUS-3290.md (hunk 2 above)
Comment (rsvoboda) [reply QUARKUS-3290.md:5]: Will update the PR ...

File: QUARKUS-3290.md (hunk 1 above)
Comment (fedinskiy) [reply QUARKUS-3290.md:22]: Maybe we should check against binaries then? Checking first N symbols for non-UTF (non ascii?) content should be trivial enough and give us a reasonable protection.
With size check  would have both false negatives (small binaries) and false positives (huge source code files)

**Pull request merged**